src/cli/subcommands/sync_cmd.rs

```diff
@@ -150,3 +165,47 @@ impl SyncCommands {
         }
     }
 }
+
+fn format_tipset_cids(cids: &str) -> &str {
+    if cids.is_empty() {
+        "[]"
+    } else {
+        cids
+    }
+}
+
+/// Check if the snapshot download is in progress, if it is then wait till the snapshot download is done
+async fn check_snapshot_progress(client: &rpc::Client) -> anyhow::Result<()> {
+    let mut interval = time::interval(Duration::from_secs(5));
+    let mut stdout = stdout();
+    loop {
+        interval.tick().await;
+        let progress_state = client.call(SyncSnapshotProgress::request(())?).await?;
+        match progress_state {
+            SnapshotProgressState::InProgress { message } => {
+                println!("🌳 Snapshot download in progress: {}", message);
+                write!(
+                    stdout,
+                    "\r{}{}",
+                    anes::ClearLine::All,
+                    anes::MoveCursorUp(1)
+                )?;
+                continue;
+            }
+            SnapshotProgressState::Completed => {
+                write!(
+                    stdout,
+                    "\r{}{}",
+                    anes::ClearLine::All,
+                    anes::MoveCursorUp(1)
+                )?;
+                println!("\n✅ Snapshot download completed! Chain will start syncing shortly (retry sync status command in 5 seconds)...");
```

Comment:
@akaladarshi it would be better to display sync status once snapshot is downloaded and chain starts syncing instead of asking user to re-run after 5 seconds and exiting

Reply:
I thought about that too but the things is we have to wait for some time before the node actually starts to sync after snapshot downloaded and wasn't sure about how much time we have to wait (actually depends on the node), so I just asked user to wait and retry.